### PR TITLE
[DDW-534, DDW-455] Update to Electron 3 and implement "instance-lock" feature, Improve lock-file UX

### PR DESCRIPTION
**Pull request by DominikGuzei**
**Please NOTE: this depends on #1219 so please merge that first**

This PR updates to Electron 3 which includes the `requireSingleInstanceLock` feature, which hopefully works more reliable than our lock file logic.

This PR improves the lock file mechanism UX when the user tries to launch a secondary Daedalus instance. Now it focuses the first window after exiting the second one (originally implemented in https://github.com/input-output-hk/daedalus/pull/1226 - but never merged).

---

### Basics

- [ ] PR is updated to the most recent version of target branch (and there are no conflicts)
- [ ] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance tests are passing (`yarn run test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn run dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn run package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn run flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn run lint`)
- [ ] Text changes are proofread and approved (Jane Wild)
- [ ] There are no missing translations (running `yarn run manage:translations` produces no changes)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn run storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly documented and commented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-rendering
- [ ] Any code that only works in Electron is neatly separated from components

### Testing
- [ ] New feature / change is covered by acceptance tests
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature / change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

**Comment by DominikGuzei** [comment]
@DmitriiGaico i tested this on Windows and it worked correctly for me! 🎉

**Comment by nikolaglumac** [comment]
Before merging this we need to merge this one: https://github.com/input-output-hk/daedalus/pull/1219 - for which I need approval from @darko-mijic...

File: source/main/index.js [hunk 1]
```diff
@@ -79,7 +79,7 @@ const safeExit = async () => {
 app.on('ready', async () => {
   // Make sure this is the only Daedalus instance running per cluster before doing anything else
   try {
-    await acquireDaedalusInstanceLock();
+    acquireDaedalusInstanceLock();
   } catch (e) {
     const dialogTitle = 'Daedalus is unable to start!';
```
[the remark this answers is not in the document]
Comment (DominikGuzei) [reply source/main/index.js]: Hmm … but i think that should be a separate feature, right?

**Comment by nikolaglumac** [comment]
@daniloprates I have tested the Windows installers and it looks like Daedalus doesn't start at all anymore :(
Can you please check this tomorrow? 🙏

**Comment by daniloprates** [comment]
> @daniloprates I have tested the Windows installers and it looks like Daedalus doesn't start at all anymore :(
> Can you please check this tomorrow? 🙏

@nikolaglumac little [conceptual mistake](https://github.com/input-output-hk/daedalus/pull/1229/commits/a254cd5520e261c03142b9003fc6e3b4750359af#diff-286decdfc7881f7334e73465dab511ffR157). I'll test on windows tomorrow morning.

**Comment by nikolaglumac** [comment]
@daniloprates thanks for the quick fix!
I have tested it on Windows and it works great 🎉 
The only issue is that our acceptance tests don't work anymore :(

**Pull request merged**